tool/site/client-react/src/pages/returnPage.tsx

```diff
@@ -0,0 +1,217 @@
+        </SummaryList.Row>
+      </SummaryList>
+      <Formik<ReturnFormValues> initialValues={initialValues} onSubmit={values => onSubmit(values)}>
+        {formik => <Form onSubmit={formik.handleSubmit} onReset={formik.handleReset}>
+          <Fieldset>
+            <PharmacyRadios
+              label="Pharmacy returning prescription"
+              defaultValue={initialValues.pharmacy}
+              value={formik.values.pharmacy}
+              error={formik.errors.pharmacy} />
+            <RadioField
+              name="reason"
+              label="Choose a reason for returning"
+              defaultValue={initialValues.reason}
+              fieldRadios={returnReasons} />
+          </Fieldset>
+          <ButtonList>
+            <Button type="submit">Return</Button>
+            <BackButton />
+          </ButtonList>
+        </Form>}
+      </Formik>
+    </>
+  )
+}
+
+async function sendReturn(
+  baseUrl: string,
+  returnFormValues: ReturnFormValues
+): Promise<ApiResult> {
+  const returnParameters = createReturn(returnFormValues)
+  const returnResponse = await axiosInstance.post<ApiResult>(`${baseUrl}dispense/return`, returnParameters)
+  return getResponseDataIfValid(returnResponse, isApiResult)
+}
+
+function createReturn(returnFormValues: ReturnFormValues): fhir.Task {
+  const identifier = uuid.v4()
+  const bundleIdentifier = identifier
+
+  return {
+    resourceType: "Task",
+    id: identifier,
+    identifier: [
+      {
+        system: "https://tools.ietf.org/html/rfc4122",
+        value: bundleIdentifier
+      }
+    ],
+    status: "rejected",
+    intent: "order",
+    groupIdentifier: {
+      system: "https://fhir.nhs.uk/Id/prescription-order-number",
+      value: returnFormValues.prescriptionId
+    },
+    code: {
+      coding: [
+        {
+          system: "http://hl7.org/fhir/CodeSystem/task-code",
+          code: "change",
+          display: "Change the focal resource"
+        }
+      ]
+    },
+    focus: {
+      type: "Bundle",
+      identifier: {
+        system: "https://tools.ietf.org/html/rfc4122",
+        value: bundleIdentifier
+      }
+    },
+    for: {
+      identifier: {
+        system: "https://fhir.nhs.uk/Id/nhs-number",
+        value: "9999999999"
+      }
+    },
+    authoredOn: formatCurrentDateTimeIsoFormat(),
+    owner: {
+      identifier: {
+        system: "https://fhir.nhs.uk/Id/ods-organization-code",
+        value: getReturnPharmacy(returnFormValues)
+      }
+    },
+    statusReason: {
+      coding: [
+        {
+          system: "https://fhir.nhs.uk/CodeSystem/EPS-task-dispense-return-status-reason",
+          code: returnFormValues.reason,
+          display: returnReasons.find(r => r.value === returnFormValues.reason).text
+        }
+      ]
+    }
+  }
+}
+
+function getReturnPharmacy(returnFormValues: ReturnFormValues) {
+  return returnFormValues.pharmacy === "custom"
+    ? returnFormValues.customPharmacy
```

Comment:
is the value for `customPharmacy` set anywhere? i cant see in the code. cant we just override the value for `pharmacy` in place wherever `customPharmacy` is set?

Reply:
you can see this behaviour for the PharmacyRadios component which is also used on release page, it wouldn't be a straightforward change I think, because we would be fighting against the benefits that Formik gives us